src/test/java/com/box/sdk/BoxAPIConnectionTest.java

```diff
@@ -569,48 +569,65 @@ public void getLowerScopedTokenRefreshesTheTokenIfNeededbyCallingGetAccessToken(
         List<String> scopes = new ArrayList<String>();
         scopes.add("DummyScope");
         String resource = "";
+        String sharedLink = null;
 
         when(api.getTokenURL()).thenReturn("https://api.box.com/oauth2/token");
-        when(api.getLowerScopedToken(scopes, resource)).thenCallRealMethod();
+        when(api.getLowerScopedToken(scopes, resource, sharedLink)).thenCallRealMethod();
         try {
-            api.getLowerScopedToken(scopes, resource);
+            api.getLowerScopedToken(scopes, resource, sharedLink);
         } catch (RuntimeException e) {
             //Ignore it
         }
         verify(api).getAccessToken();
     }
 
     @Test
-    @Category(UnitTest.class)
-    public void getLowerScopedTokenWithNullResource() {
-        BoxAPIConnection api = mock(BoxAPIConnection.class);
+    @Category(IntegrationTest.class)
+    public void getLowerScopedToken() {
+        final String originalAccessToken = TestConfig.getAccessToken();
+        BoxAPIConnection api = new BoxAPIConnection(originalAccessToken);
 
         List<String> scopes = new ArrayList<String>();
-        scopes.add("DummyScope");
+        scopes.add("item_preview");
+        scopes.add("item_content_upload");
         String resource = null;
+        String sharedLink = null;
 
-        when(api.getTokenURL()).thenReturn("https://api.box.com/oauth2/token");
-        when(api.getLowerScopedToken(scopes, resource)).thenCallRealMethod();
-        try {
-            api.getLowerScopedToken(scopes, resource);
-        } catch (RuntimeException e) {
-            //Ignore it
-        }
-        verify(api).getAccessToken();
+        ScopedToken token = api.getLowerScopedToken(scopes, resource, sharedLink);
+        assertThat(token, notNullValue());
+        assertThat(token.getAccessToken(), notNullValue());
     }
 
     @Test
     @Category(IntegrationTest.class)
-    public void getLowerScopedTokenWorks() {
+    public void getLowerScopedTokenForResource() {
```

Comment:
I think we can also add a unit test here with the intent to force it down the new if branches in boxapiconnection class. We can just have it return a string but we can assert that request to downscope token has all the parameter in the json body